crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -180,20 +299,872 @@ fn __num_basis_gates(basis_b: f64, basis_fidelity: f64, unitary: MatRef<c64>) ->
     traces
         .into_iter()
         .enumerate()
-        .map(|(idx, trace)| (idx, trace_to_fid(&trace) * basis_fidelity.powi(idx as i32)))
+        .map(|(idx, trace)| (idx, trace.trace_to_fid() * basis_fidelity.powi(idx as i32)))
         .min_by(|(_idx1, fid1), (_idx2, fid2)| fid2.partial_cmp(fid1).unwrap())
         .unwrap()
         .0
 }
 
-/// Average gate fidelity is :math:`Fbar = (d + |Tr (Utarget \\cdot U^dag)|^2) / d(d+1)`
-/// M. Horodecki, P. Horodecki and R. Horodecki, PRA 60, 1888 (1999)
-fn trace_to_fid(trace: &c64) -> f64 {
-    (4.0 + trace.faer_abs2()) / 20.0
+/// A good approximation to the best value x to get the minimum
+/// trace distance for :math:`U_d(x, x, x)` from :math:`U_d(a, b, c)`.
+fn closest_partial_swap(a: f64, b: f64, c: f64) -> f64 {
+    let m = (a + b + c) / 3.;
+    let [am, bm, cm] = [a - m, b - m, c - m];
+    let [ab, bc, ca] = [a - b, b - c, c - a];
+    m + am * bm * cm * (6. + ab * ab + bc * bc + ca * ca) / 18.
+}
+
+fn rx_matrix(theta: f64) -> Array2<Complex64> {
+    let half_theta = theta / 2.;
+    let cos = Complex64::new(half_theta.cos(), 0.);
+    let isin = Complex64::new(0., -half_theta.sin());
+    array![[cos, isin], [isin, cos]]
+}
+
+fn ry_matrix(theta: f64) -> Array2<Complex64> {
+    let half_theta = theta / 2.;
+    let cos = Complex64::new(half_theta.cos(), 0.);
+    let sin = Complex64::new(half_theta.sin(), 0.);
+    array![[cos, -sin], [sin, cos]]
+}
+
+fn rz_matrix(theta: f64) -> Array2<Complex64> {
+    let ilam2 = Complex64::new(0., 0.5 * theta);
+    array![
+        [(-ilam2).exp(), Complex64::new(0., 0.)],
+        [Complex64::new(0., 0.), ilam2.exp()]
+    ]
+}
+
+const DEFAULT_FIDELITY: f64 = 1.0 - 1.0e-9;
+const C1_IM: Complex64 = Complex64::new(0.0, 1.0);
+
+#[derive(Clone, Debug, Copy)]
+#[pyclass(module = "qiskit._accelerate.two_qubit_decompose")]
+enum Specialization {
+    General,
+    IdEquiv,
+    SWAPEquiv,
+    PartialSWAPEquiv,
+    PartialSWAPFlipEquiv,
+    ControlledEquiv,
+    MirrorControlledEquiv,
+    // These next 3 gates use the definition of fSim from eq (1) in:
+    // https://arxiv.org/pdf/2001.08343.pdf
+    SimaabEquiv,
+    SimabbEquiv,
+    SimabmbEquiv,
```

Comment:
```suggestion
    fSimaabEquiv,
    fSimabbEquiv,
    fSimabmbEquiv,
```
The conventional name for the gate is "fSim".

Reply:
I did this because the lint rules didn't like the lower case "f". But I agree it's better to use the proper name; I'll just suppress that lint rule

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11946/commits/ade9c8010bc358f6abe2bb8ee180790a24be2dd4